### ISSUE TITLE
feat(bridgebuilder): consensus voting can suppress single-model security true-positives

### ISSUE THREAD

**Issue by deep-name** [0xHoneyJar/loa#810]
## Summary

Bridgebuilder's `HIGH_CONSENSUS` classification requires ≥2 reviewer models to flag the same finding independently. **Security findings are exactly where this voting can fail-safe into false-negative**: one careful model spots a real issue, the second model focuses elsewhere and misses it, and the finding lands in the `DISPUTED` bucket where it competes for attention with low-confidence noise.

The headline "0 BLOCKER, 0 HIGH_CONSENSUS" output of a multi-model BB run is technically correct but doesn't carry the right operational weight when a single model has surfaced a real security concern at, say, 0.55 confidence on the Security dimension — exactly the shape that will get triaged as "noise" by a hurried operator.

## Empirical instance

PR #803 (cycle-102 sprint-1), Bridgebuilder iteration 5 (2026-05-09):

```
Models: anthropic/claude-opus-4-7, openai/gpt-5.5-pro, google/gemini-3.1-pro-preview (errored)

| Classification | Count |
|---|---|
| HIGH_CONSENSUS | 0 |
| DISPUTED | 5 |
| BLOCKER | 0 |
| LOW_VALUE | 18 |
| Unique perspectives | 23 |
```

**FIND-005** (Anthropic single-model, 0.55 confidence, [Security] dimension):

> "Raw exception strings and stack traces commonly contain prompt fragments, URLs, auth headers, or tokens. A length cap is not redaction. The redacted-envelope contract leaks at the UNKNOWN branch — exactly the branch most likely to fire in novel failure modes."

This was a real finding against `model-error.schema.json`'s `original_exception` field. The schema description handwaves redaction to "emitter responsibility + downstream lint catches drift" — but (a) the audit chain is hash-chained immutable per cycle-098, so any leak is permanent; (b) the "downstream lint" was aspirational; (c) the UNKNOWN branch is the worst time to leak because the emitter doesn't know what to redact.

OpenAI GPT-5.5-pro reviewed the same diff and **did not** flag the redaction gap. It focused instead on test-quality findings (static-grep tests, format_checker drift). Both reviews were thorough; they just looked at different surfaces.

Without skeptical re-read of the DISPUTED bucket, the headline-number triage would have shipped FIND-005 as "noise" and the redaction gap would have stayed open until the first emitter wired up audit_emit and started leaking.

The full re-classification is in `grimoires/loa/cycles/cycle-102-model-stability/sprint.md` Sprint 1B section as T1B.1 (HIGH).

## Why consensus voting can't catch this naturally

The fundamental tension:
- **HIGH_CONSENSUS** is a defense against single-model hallucination — both models flagged it, so it's probably real.
- **Security findings** are often asymmetric: one careful model (e.g., one with stronger SecOps training corpus) sees the threat model; the other doesn't.

For non-security domains, this is fine — single-model findings are usually noise. **For security domains, single-model findings are often signal**, because the threat-model intuition is rare.

## Proposed mitigation

Three options, lowest-risk first:

### Option A — Dimension-weighted classification (recommended)

When a finding lands in `DISPUTED` AND has `dimension: "[Security]"` or category in a hard-coded sensitive set (`Security`, `Validation`, `Authorization`, `Audit`, `Crypto`), promote its display priority in the consensus output. Don't change the bucket label; just surface it above LOW_VALUE in the rendered review.

Implementation sketch (TS):

```ts
// in core/scoring.ts or wherever consensus is rendered
const SECURITY_DIMENSIONS = new Set(['[Security]', 'Security', 'Validation', 'Authorization', 'Audit', 'Crypto']);

function isSecurityDisputed(finding: Finding): boolean {
  return finding.classification === 'DISPUTED' && SECURITY_DIMENSIONS.has(finding.dimension);
}

// at render time, surface security-disputed findings in their own section
```

The rendered review would then look like:

```markdown
| HIGH_CONSENSUS | 0 |
| BLOCKER | 0 |
| DISPUTED (security-relevant) | 1  ← surfaced separately
| DISPUTED (other) | 4 |
| LOW_VALUE | 18 |
```

### Option B — Lower the consensus threshold for security dimensions

Currently HIGH_CONSENSUS = "≥2 models flagged independently." For Security findings only, allow `≥1 model with confidence ≥0.5` to be classified as HIGH_CONSENSUS. This is more aggressive and could increase noise; pair with operator-side suppression rules (`.reviewignore` security carve-outs).

### Option C — Consensus + skeptical-second-pass

After the parallel review, run a second pass that asks one model to specifically argue *against* dismissing each DISPUTED finding. The second-pass output then determines whether the finding is genuine-disputed or surfaces-as-elevated.

This is heavier on tokens but matches the established `enrichment` writer pattern.

## Acceptance criteria

- A repro fixture: a synthetic finding marked Security/medium/single-model triggers the new classification path.
- Documentation update: `grimoires/bridgebuilder/BEAUVOIR.md` (or wherever the persona-level scoring guidance lives) explicitly notes that Security DISPUTED findings warrant operator review even when classification labels are not BLOCKER.
- Backward compatibility: existing HIGH_CONSENSUS / BLOCKER consumers see no change to existing buckets.

## Origin

- BB iter-5 on PR #803 (cycle-102 sprint-1), 2026-05-09
- Operator skepticism that "0 BLOCKER" patterns warrant scrutiny — caught the FIND-005 elevation in real time
- Decision Log entry: `grimoires/loa/NOTES.md` 2026-05-09 (cycle-102 Sprint 1A — ship/no-ship + AC deferrals)
- Cross-reference: cycle-102 sprint.md Sprint 1B T1B.1 (the project-specific fix that this framework issue would generalize)

## Out of scope

- Changing the `HIGH_CONSENSUS` definition for non-security dimensions (would change behavior for every BB consumer)
- Token-cost increase from skeptical-second-pass (Option C) — only viable if multi-model budget is already on the higher tier

**Comment by deep-name** [comment]
Cross-link: filed #811 (Marasa-Trois empirical falsification) which is the dual of this issue. Where #810 addresses *consensus suppressing true-positives* (one model right, others wrong), #811 addresses *consensus surfacing false-positives* (multiple models confidently-wrong from correlated training data). Both converge on the need for a non-LLM ground-truth oracle.

**Comment by deep-name** [comment]
## Cycle-104 cross-reference — voice-drop wiring (T2.8) addresses an adjacent class

This issue (consensus voting suppresses single-model security true-positives) is **NOT directly fixed** by cycle-104 work, but the cycle-104 voice-drop wiring (T2.8) addresses an adjacent failure class worth cross-referencing:

### What T2.8 fixes (NOT this issue)

When a voice's within-company fallback chain exhausts (cheval exit 12 = `CHAIN_EXHAUSTED`), the orchestrator DROPS that voice from consensus rather than substituting another company's model. The `consensus.voice_dropped` event surfaces in the trajectory log; consensus proceeds with the remaining voices.

This closes the cycle-102 T1B.4 cross-company-substitution anti-pattern — but the substitution happened at the CHAIN level (post-exhaustion), not at the CONSENSUS-VOTING level.

### What this issue is about (NOT fixed by T2.8)

Single-model true-positives that the consensus algorithm suppresses because other models didn't surface them. This is a CONSENSUS-LOGIC issue, not a chain-exhaustion issue. The current consensus algorithm treats "only 1 of N models flagged" as low-confidence; that's the bug — security true-positives are often single-model by nature (one model's prompt-injection alarm, another's missing-RLS detection, etc.).

### Suggested fix paths (still outstanding)

1. **Severity-aware threshold** — `BLOCKING` / `CRITICAL` severity findings bypass the multi-model-consensus floor (single-model claim is enough)
2. **Class-aware threshold** — security-class findings (taxonomy-tagged) bypass the floor
3. **Single-model-true-positive lens** — already documented in `feedback_zero_blocker_demotion_pattern.md`; could be promoted to BB / flatline scoring engine

Cycle-104 voice-drop does NOT close this issue. Leaving OPEN.

### Related

- #807 (multi-model fallback approves issues single-model misses) — same class
- `feedback_zero_blocker_demotion_pattern.md` operator memory (`grimoires/loa/memory/` — operator-local)

**Comment by deep-name** [comment]
**Decision (operator delegated 2026-06-13): executing Option A, scoped exactly per the wave plan — render-time surfacing only; B/C stay deferred.**

The calculus has strengthened since filing: (a) cycle-104 voice-drop means chain exhaustion silently degrades runs toward single-voice, making the single-model-MEDIUM→DISPUTED path more common, (b) post-pr-triage.sh:248 buckets DISPUTED with deferred/HITL in the autonomous loop, so render-time surfacing is the only visibility hook there, (c) KF-004 (recurrence ≥24) made silent finding-suppression the wave's defining theme.

Implementation notes from verification: the gap is narrower than the headline — single-model CRITICAL/BLOCKER already escalates (scoring.ts:146-148) and single-model HIGH already clears HIGH_CONSENSUS (score 800 ≥ 700); the suppression window is exactly single-model MEDIUM (score 500), the FIND-005 shape. The finding schema has no `dimension` field — match will be normalize-then-match on `category` (security/validation/authorization/audit/crypto, bracket-stripped, case-insensitive). Both render paths (formatConsensusSummary AND formatEnrichedConsensusSummary) get the split row + surfaced section; classification enums and bucket counts stay untouched (post-pr-triage.sh awk-greps raw strings). dist/ rebuild + freshness manifest in the same PR (check-bb-dist-fresh gate).

Queued as a micro-sprint (cites the empirical FIND-005 suppression on PR #803 as the observed failure) behind the HELD-PR fixes in flight.
